### ISSUE TITLE
no-trailing-spaces does not properly handle skipBlankLines

### ISSUE THREAD

**Issue by rabbitfang** [eslint/eslint#2575]
When running
```
eslint --reset --rule 'no-trailing-spaces: [2, {skipBlankLines:true}]' test.js
```

with `test.js` as (where `_` is a space and `>` is a tab character)
```javascript
var a = 'bar';__
_
>
```

outputs the following:
```
test.js
  2:0  error  Trailing spaces not allowed  no-trailing-spaces
  3:0  error  Trailing spaces not allowed  no-trailing-spaces

✖ 2 problems (2 errors, 0 warnings)
```

which is incorrect. The expected outcome is
```
test.js
  1:14  error  Trailing spaces not allowed  no-trailing-spaces

✖ 1 problem (1 error, 0 warnings)
```

This is being caused by the [regex used for detecting trailing whitespace](https://github.com/eslint/eslint/blob/master/lib/rules/no-trailing-spaces.js#L32).

**Comment by ilyavolodin** [comment]
Which version of ESLint are you using?

**Comment by ilyavolodin** [comment]
I was able to reproduce this with the default parser and latest version of ESLint. Seems like a bug to me.

**Comment by pmcelhaney** [comment]
That [regex](https://github.com/eslint/eslint/blob/master/lib/rules/no-trailing-spaces.js#L13) is a little weird.

    var BLANK_PREFIX = "[^( |\t)]"

It could be reduced to this. Not sure if that makes any difference. 

    var BLANK_PREFIX = "[^ \t]"

**Comment by gcochard** [comment]
I'll take this on.
